File: src/main/java/net/softbell/bsh/filter/ControllerFilter.java
package net.softbell.bsh.filter;

import java.security.Principal;

import org.springframework.ui.Model;
import org.springframework.web.servlet.ModelAndView;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 컨트롤러가 동작하기 전 필터 클래스
 */
public class ControllerFilter {
	// Static Global Field
	public static final String G_SERVICE_ERROR_REDIRECT_URL = "redirect:/error/service";
	public static final String G_LOGOUT_REDIRECT_URL = "redirect:/logout";
	public static final String G_BAN_REDIRECT_URL = "redirect:/denied";
		
	public void FilterModelPrincipal(Model model, Principal principal)
	{
		// Username
		if (principal != null)
			model.addAttribute("USER_NAME", principal.getName());
	}
	
	public ModelAndView FilterModelAndViewPrincipal(ModelAndView model, Principal principal)
	{
		// Username
		if (principal != null)
			model.addObject("USER_NAME", principal.getName());
		
		return model;
	}
}

File: src/main/kotlin/net/softbell/bsh/filter/ControllerFilter.kt
package net.softbell.bsh.filter

import org.springframework.ui.Model
import org.springframework.web.servlet.ModelAndView
import java.security.Principal

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 컨트롤러가 동작하기 전 필터 클래스
 */
class ControllerFilter {
    fun FilterModelPrincipal(model: Model, principal: Principal?) {
        // Username
        if (principal != null) model.addAttribute("USER_NAME", principal.name)
    }

    fun FilterModelAndViewPrincipal(model: ModelAndView, principal: Principal?): ModelAndView {
        // Username
        if (principal != null) model.addObject("USER_NAME", principal.name)
        return model
    }

    companion object {
        // Static Global Field
        const val G_SERVICE_ERROR_REDIRECT_URL = "redirect:/error/service"
        const val G_LOGOUT_REDIRECT_URL = "redirect:/logout"
        const val G_BAN_REDIRECT_URL = "redirect:/denied"
    }
}